packages/read-emails/frontend/react/src/EmailDetail.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import EmailIllustration from './components/icons/illustration-email.svg';
+import ChevronDown from './components/icons/icon-chevron-down.svg';
+import { formatPreviewDate } from './utils/date.js';
+import { cleanEmailBody } from './utils/email.js';
+
+function EmailDetail({ selectedEmail, userEmail }) {
+  const [emailSender, setEmailSender] = useState('');
+  const [emailReceivers, setEmailReceivers] = useState('');
+  const [showParticipants, setShowParticipants] = useState(false);
+
+  useEffect(() => {
+    if (selectedEmail?.from?.length) {
+      setEmailSender(selectedEmail?.from?.[0]);
+    } else {
+      setEmailSender('');
+    }
+    getReceivers();
+  }, [selectedEmail]);
+
+  const getReceivers = () => {
+    let receiversStr = '';
+    const receiverList = [];
+
+    if (selectedEmail?.to?.length) {
+      for (let i = 0; i < selectedEmail.to.length; i++) {
+        if (i === 2) {
+          receiverList.push(`+${(selectedEmail.to.length - 2).toString()}`);
+          break;
+        }
+
+        const to = selectedEmail.to[i];
+        receiverList.push(to.email === userEmail ? 'Me' : to.name || to.email);
```

Comment:
This looks odd.  And because it breaks after checking the first two elements, if your email address is in the `to:` field but isn't within the first two elements of the array it won't show "Me".

Can we check to see if `userEmail` is in the array and then just take a slice of the remaining elements versus looping just to break out after two iterations?

Reply:
I think user email doesn't need to show up if it's not the first two elements of the array. 
It's the same behaviour as Gmail.